src/posix.c

```diff
@@ -266,7 +303,78 @@ int p_munmap(git_map *map)
 	assert(map != NULL);
 	git__free(map->data);
 
+	/* Initializing will help debug use-after-free */
+	map->len = 0;
+	map->data = NULL;
+
 	return 0;
 }
 
+ssize_t p_pwrite(git_file fd, const void *buf, size_t nbyte, git_off_t offset)
+{
+#ifdef GIT_WIN32
+	/* FIXME: Implement using Win32 overlapped IO */
+
+	ssize_t ret;
+	int nfd = dup(fd);
+	
+	(void) ((ret = p_lseek(nfd, offset, SEEK_SET) < 0 ? -1 : 0) ||
```

Comment:
this style doesn't seem to appear anywhere else in the codebase. would it be possible to use the more common

```c
if ((error = ...) < 0)
  goto cleanup;
if ((error = ...) < 0)
  goto cleanup;
...
cleanup:
  close(nfd);
  return error;
```